### DISCUSSION TITLE
Ability to schedule a recurring task per node

### DISCUSSION THREAD

**Discussion by reda-alaoui** [kagkarlsson/db-scheduler#818] [General]
I have a fixed delay recurring task for which each excution ocurrence must execute on all nodes. Those executions do not need to be synced between the nodes.
So I thought I would schedule the task on each node with a name unique to each node.
Suppose I have 2 nodes. After 14 days, will node 1 removes the execution of node 2 task?

**Comment by eRfO** [comment]
@kagkarlsson I have the same need to @reda-alaoui , it would be possible to add in removeExecutions a filter that remove only the not running unresolved tasks?

Thanks

**Comment by kagkarlsson** [comment]
This is a use-case not very well suited for db-scheduler. When I have something I need to run on all instances of my app, I usually go with a standard Java `ScheduledExecutorService`. It has a `scheduleWithFixedDelay` method that does what you want

**Comment by kagkarlsson** [comment]
Is there a reason you cannot use a regular `ScheduledExecutorService` for this particular task, and db-scheduler for the rest?

**Comment by reda-alaoui** [comment]
I considered this option.
But I would prefer using `db-scheduler` for this case to keep using its abstractions. Among the tasks, I have some of them that need to be cluster aware for one schedule and local for another. These tasks already use `db-scheduler` objects such as `TaskInstance`.
Another way of achieving that would be to have a separate `Scheduler` instance running on top of an in-memory database like H2.

**Comment by eRfO** [comment]
In my case the task per node use is related to cluster view, to monitor which nodes are currently active. I'm taking advantage of the heartbeat feature of the db-scheduler to achieve this.
